packages/espressocash_app/lib/utils/solana_pay.dart

```diff
@@ -1,20 +1,26 @@
+import 'package:espressocash_common/espressocash_common.dart';
 import 'package:solana/solana_pay.dart';
 
 import '../config.dart';
 
 SolanaPayRequest? tryParseSolanaPayRequest(Uri link) {
-  final linkWithCorrectScheme = link.scheme == 'https' &&
-          link.host == espressoCashLinkDomain &&
-          link.queryParameters['t'] == 'solanapay' &&
-          link.queryParameters['recipient'] != null
-      ? Uri(
-          scheme: 'solana',
-          path: link.queryParameters['recipient'],
-          queryParameters: {...link.queryParameters}
-            ..remove('t')
-            ..remove('recipient'),
-        )
-      : link;
+  final isEcRequest =
+      link.scheme == 'https' && link.host == espressoCashLinkDomain;
 
-  return SolanaPayRequest.tryParse(linkWithCorrectScheme.toString());
+  if (!isEcRequest) return SolanaPayRequest.tryParse(link.toString());
+
+  final Map<String, String> queryParams = link.queryParameters;
+
+  final url = Uri(
+    scheme: 'solana',
+    path: queryParams['recipient'] ?? queryParams['r'],
+    queryParameters: {
+      'amount': queryParams['amount'] ?? queryParams['a'],
+      'reference': queryParams['reference'] ?? queryParams['p'],
+      'label': queryParams['label'] ?? queryParams['l'],
+      'spl-token': queryParams['spl-token'] ?? Token.usdc.address,
```

Comment:
We cannot do it like that. In Solana Pay, missing `spl-token` means it's SOL transaction.